src/main/java/pro/trevor/tankgame/state/board/unit/tank/GenericTank.java

```diff
@@ -0,0 +1,158 @@
+package pro.trevor.tankgame.state.board.unit.tank;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import pro.trevor.tankgame.rule.type.IPlayerElement;
+import pro.trevor.tankgame.rule.type.ITickElement;
+import pro.trevor.tankgame.state.board.IMovable;
+import pro.trevor.tankgame.state.board.IPositioned;
+import pro.trevor.tankgame.state.board.Position;
+import pro.trevor.tankgame.state.board.unit.IUnit;
+import pro.trevor.tankgame.state.board.unit.tank.status.IAttributeStatus;
+import pro.trevor.tankgame.state.board.unit.tank.status.IStatus;
+import pro.trevor.tankgame.state.board.unit.tank.status.IStatusDecoder;
+
+import java.util.*;
+
+public class GenericTank<E extends Enum<E> & IAttribute> implements IUnit, IMovable, IPositioned, ITickElement, IPlayerElement {
```

Comment:
I think this class should be split into generic entity and generic tank. All entities have attributes so at least the type/attribute part should be broken out.